apps/financroo-tpp/clients.go

```diff
@@ -45,16 +48,44 @@ type ConsentClient interface {
 
 func (c *Clients) RenewAccountsToken(ctx context.Context, bank ConnectedBank) (*oauth2Models.TokenResponse, error) {
 	var (
-		resp *oauth2.TokenOK
-		err  error
+		resp      oauth2.TokenOK
+		request   *http.Request
+		response  *http.Response
+		assertion string
+		body      []byte
+		err       error
 	)
 
-	if resp, err = c.AcpAccountsClient.Oauth2.Oauth2.Token(
-		oauth2.NewTokenParamsWithContext(ctx).
-			WithClientID(&c.AcpAccountsClient.Config.ClientID).
-			WithGrantType("refresh_token").
-			WithRefreshToken(&bank.RefreshToken)); err != nil {
-		return nil, errors.Wrapf(err, "can't renew access token for a bank: %s", bank.BankID)
+	values := url.Values{
+		"client_id":     {c.AcpAccountsClient.Config.ClientID},
+		"grant_type":    {"refresh_token"},
+		"refresh_token": {bank.RefreshToken},
+	}
+
+	if c.AcpAccountsClient.Config.AuthMethod == acpclient.PrivateKeyJwtAuthnMethod {
+		if assertion, err = c.AcpAccountsClient.GenerateClientAssertion(); err != nil {
+			return nil, err
+		}
+		values.Add("client_assertion_type", "urn:ietf:params:oauth:client-assertion-type:jwt-bearer")
```

Comment:
It seems that this code will be repeated every time you want to use the private_key authn method.
We could introduce some helper methods for token endpoint, similar to AuthorizeOptions:

```
func WithPKCE() AuthorizeOption {
	return authorizeHandler(func(c *Client, v url.Values, csrf *CSRF) (err error) {
		if csrf.Verifier, err = randomString(VerifierLength); err != nil {
			return errors.Wrapf(err, "failed to generate random verifier for PKCE authentication")
		}

		hash := sha256.New()

		if _, err = hash.Write([]byte(csrf.Verifier)); err != nil {
			return errors.Wrapf(err, "failed to hash PKCE verifier")
		}

		v.Set("code_challenge", base64.RawURLEncoding.WithPadding(base64.NoPadding).EncodeToString(hash.Sum([]byte{})))
		v.Set("code_challenge_method", "S256")

		return nil
	})
}
```

This can be done in a separate PR